Library/Homebrew/archive.rb

```diff
@@ -0,0 +1,185 @@
+      opoo msg
+      return
+    end
+
+    md5_base64 = Digest::MD5.base64digest(File.read(local_file))
+    url = "https://#{@archive_item}.s3.us.archive.org/#{directory}/#{remote_file}"
+    args = ["--upload-file", local_file, "--header", "Content-MD5: #{md5_base64}"]
+    args << "--fail" unless warn_on_error
+    result = T.unsafe(self).open_api(url, *args)
+    return if result.success? && result.stdout.exclude?("Error")
+
+    msg = "Bottle upload failed: #{result.stdout}"
+    raise msg unless warn_on_error
+
+    opoo msg
+  end
+
+  sig { params(url: String).returns(T::Boolean) }
+  def stable_mirrored?(url)
+    headers, = curl_output("--connect-timeout", "15", "--location", "--head", url)
+    status_code = headers.scan(%r{^HTTP/.* (\d+)}).last.first
+    status_code.start_with?("2")
+  end
+
+  sig {
+    params(formula:       Formula,
+           directory:     String,
+           warn_on_error: T::Boolean).returns(String)
+  }
+  def mirror_formula(formula, directory: "mirror", warn_on_error: false)
+    formula.downloader.fetch
+
+    filename = ERB::Util.url_encode(formula.downloader.basename)
+    destination_url = "https://archive.org/download/#{@archive_item}/#{directory}/#{filename}"
+
+    odebug "Uploading to #{destination_url}"
+
+    upload(
+      formula.downloader.cached_location,
+      directory:     directory,
+      remote_file:   filename,
+      warn_on_error: warn_on_error,
+    )
+
+    destination_url
+  end
+
+  # Gets the MD5 hash of the specified remote file.
+  #
+  # @return the hash, the empty string (if the file doesn't have a hash), nil (if the file doesn't exist)
+  sig { params(directory: String, remote_file: String).returns(T.nilable(String)) }
+  def remote_md5(directory:, remote_file:)
+    url = "https://#{@archive_item}.s3.us.archive.org/#{directory}/#{remote_file}"
+    result = curl_output "--fail", "--silent", "--head", "--location", url
+    if result.success?
+      result.stdout.match(/^ETag: "(\h{32})"/)&.values_at(1)&.first || ""
+    else
+      raise Error if result.status.exitstatus != 22 && result.stderr.exclude?("404 Not Found")
+
+      nil
+    end
+  end
+
+  sig { params(directory: String, filename: String).returns(String) }
+  def file_delete_instructions(directory, filename)
+    <<~EOS
+      Run:
+        curl -X DELETE -H "Authorization: AWS $HOMEBREW_ARCHIVE_KEY" https://#{@archive_item}.s3.us.archive.org/#{directory}/#{filename}
+      Or run:
+        ia delete #{@archive_item} #{directory}/#{filename}
+    EOS
+  end
+
+  sig {
+    params(bottles_hash:  T::Hash[String, T.untyped],
+           warn_on_error: T.nilable(T::Boolean)).void
+  }
+  def upload_bottles(bottles_hash, warn_on_error: false)
+    bottles_hash.each do |_formula_name, bottle_hash|
+      directory = bottle_hash["bintray"]["repository"]
+      bottle_count = bottle_hash["bottle"]["tags"].length
+
+      bottle_hash["bottle"]["tags"].each do |_tag, tag_hash|
```

Comment:
```suggestion
      bottle_hash["bottle"]["tags"].each_value do |tag_hash|
```

Reply:
Nice catch. Rubocop should have an audit for this case.